Lib/test/test_shutil.py

```diff
@@ -1848,6 +1848,23 @@ def test_pathext(self):
             rv = shutil.which(program, path=self.temp_dir)
             self.assertEqual(rv, temp_filexyz.name)
 
+    @unittest.skipUnless(sys.platform == "win32", 'test specific to Windows')
+    def test_pathext_with_empty_str(self):
+        ext = ".xyz"
+        temp_filexyz = tempfile.NamedTemporaryFile(dir=self.temp_dir,
+                                                   prefix="Tmp2", suffix=ext)
+        os.chmod(temp_filexyz.name, stat.S_IXUSR)
```

Comment:
Python's implementation of `chmod` for Windows just sets or clears the readonly file attribute depending on the `S_IWRITE` (`S_IWUSR`) bit. Technically, this is wrong because attributes aren't permissions, and it's inconsistent with handling the "immutable" file attribute in Unix, but it's also been like this for decades. Anyway, without `S_IWUSR` this call just makes the file readonly.

Reply:
I copied the test from `test_pathext` (https://github.com/python/cpython/blob/master/Lib/test/test_shutil.py#L1835 ), would it make sense to just make a second `assert` there for this edge case?